js/ui/scheduler/ui.scheduler.recurrence_editor.js

```diff
@@ -199,373 +195,311 @@ const RecurrenceEditor = Editor.inherit({
+
+                    const localDaysNames = dateLocalization.getDayNames('abbreviated');
+                    const dayNames = days.slice(firstDayOfWeek).concat(days.slice(0, firstDayOfWeek));
+
+                    const itemsButtonGroup = localDaysNames.slice(firstDayOfWeek).concat(localDaysNames.slice(0, firstDayOfWeek)).map((item, index) => { return { text: item, key: dayNames[index] }; });
+
+                    this._$repeatOnWeek = $('<div>').addClass(RECURRENCE_BUTTON_GROUP).appendTo(itemElement);
+
+                    this._weekEditor = this._createComponent(this._$repeatOnWeek, ButtonGroup, {
+                        items: itemsButtonGroup,
+                        field: 'byday',
+                        selectionMode: 'multiple',
+                        selectedItemKeys: byDay,
+                        keyExpr: 'key',
+                        onSelectionChanged: (e) => {
+                            const selectedKeys = e.component.option('selectedItemKeys');
+                            this._recurrenceRule.makeRule('byday', selectedKeys);
+                            this._changeEditorValue();
+                        }
+                    });
+                },
+                visible: freq === 'weekly',
+                label: {
+                    visible: false
+                }
+            },
+            {
+                dataField: 'bymonth',
+                editorType: 'dxSelectBox',
+                editorOptions: {
+                    field: 'bymonth',
+                    items: months,
+                    value: this._monthOfYearByRules(),
+                    width: recurrentEditorSelectBoxWidth,
+                    displayExpr: 'text',
+                    valueExpr: 'value',
+                    elementAttr: {
+                        class: MONTH_OF_YEAR
+                    },
+                    onValueChanged: (args) => this._valueChangedHandler(args)
+                },
+                visible: freq === 'yearly',
+                label: {
+                    visible: false
+                }
+            },
+            {
+                dataField: 'bymonthday',
+                editorType: 'dxNumberBox',
+                editorOptions: {
+                    min: 1,
+                    max: 31,
+                    width: recurrentEditorNumberBoxWidth,
+                    field: 'bymonthday',
+                    showSpinButtons: true,
+                    useLargeSpinButtons: false,
+                    value: this._dayOfMonthByRules(),
+                    onValueChanged: (args) => this._valueChangedHandler(args),
+                    elementAttr: {
+                        class: DAY_OF_MONTH
+                    }
+                },
+                visible: freq === 'monthly' || freq === 'yearly',
+                label: {
+                    visible: false
+                }
+            }
+        ];
+    },
+
+    _getRepeatEndItems: function() {
```

Comment:
Is it possible to create this items without template by using form items?

Reply:
RadioGroup is a predefined type of editor for Form and thus I can set itemTemplate in editor options. I will correct it!